include/deal.II/base/table.h

```diff
@@ -2367,6 +2624,170 @@ Table<2,T>::n_cols () const
 
 
 
+//---------------------------------------------------------------------------
+namespace TransposeTableIterators
+{
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase()
+    :
+    object(nullptr),
+    linear_index(std::numeric_limits<decltype(linear_index)>::max())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const object_pointer_type table)
+    :
+    object(table),
+    // TODO why can't this be object->values.size()?
+    linear_index(table->values.size())
```

Comment:
That's because you should have gotten a warning that the compiler will reorder these two initializations: you declare the linear index *after* object in the class declaration, so even though you write it like this, the actual initialization happens the other way around.

In fact, I think that in all examples above (the sample doc example) and the actual `AccessorBase` class, you should declare `object` before `linear_index`: The latter is sub-ordinate to the former, so it makes sense to declare them in that order.

Reply:
That explains it. I am surprised I did not see compiler warnings.